lib/rubocop/lockfile.rb

```diff
@@ -1,5 +1,11 @@
 # frozen_string_literal: true
 
+begin
+  require 'bundler'
+rescue LoadError
+  nil
+end
+
```

Comment:
This should have been here all along, but feel free to pull it out of this PR, if you want to keep it super-focused on just fixing the exact bug that's reported by #12823

I put this as a second commit, so feel free to `drop` it if you'd like.

Reply:
This does not look right. The question with your checks in the other commit in this PR is not "is bundler available on the system at all" which this require does, but "is this project using bundler". By forcefully loading bundler here, you are also introducing it in cases where bundler is not used (e.g. in a simple ruby script file). Consequently, rubocop now fails because it cannot find a Gemfile.

Reply:
> Consequently, rubocop now fails because it cannot find a Gemfile.

Can you elaborate on that? I don't think that using bundler as a library imposes the requirement for a `Gemfile` to be present.

Reply:
I saw failures related to it and effectively the `defined?(Bundler)` check later is now always true given that normally it should always be possible to require bundler. I'll try to create a proper test case to verify and will open a separate issue for it if I can reproduce it properly.

Reply:
This issue is expected to be resolved in #12847, which aims to solve #12846.